ts/src/test/tests.ts

```diff
@@ -1026,9 +1027,17 @@ class testMainClass {
                 // when comparing the response we want to allow some flexibility, because a 50.0 can be equal to 50 after saving it to the json file
                 this.assertStaticError (sanitizedNewOutput === sanitizedStoredOutput, messageError, storedOutput, newOutput, assertingKey);
             } else {
-                const isBoolean = (typeof sanitizedNewOutput === 'boolean') || (typeof sanitizedStoredOutput === 'boolean');
-                const isString = (typeof sanitizedNewOutput === 'string') || (typeof sanitizedStoredOutput === 'string');
-                const isUndefined = (sanitizedNewOutput === undefined) || (sanitizedStoredOutput === undefined); // undefined is a perfetly valid value
+                const computedBool = (typeof sanitizedNewOutput === 'boolean');
+                const storedBool = (typeof sanitizedStoredOutput === 'boolean');
+                const computedString = (typeof sanitizedNewOutput === 'string');
+                const storedString = (typeof sanitizedStoredOutput === 'string');
+                const computedUndefined = (sanitizedNewOutput === undefined);
+                const storedUndefined = (sanitizedStoredOutput === undefined);
+                const shouldBeSame = (computedBool === storedBool) && (computedString === storedString) && (computedUndefined === storedUndefined);
```

Comment:
this condition is also confusing, at first glance I was assuming it was doing a thing completely different than it is doing  😅

Reply:
names might need to be better, right, but condition seemed good to me, wdyt? so it asserts that returned type matching is same for `computer` and `stored`

Reply:
so, in case of any condition deviates from `true` it means the types are differnt

Reply:
@ttodua  Yeah I think that if you rename the `computedBool` to something like `isComputedBool` it will make things a bit more clear

Reply:
@carlosmiei pushed ✌️ the build is broken by hyperliquid.